### ISSUE TITLE
Crash when calling TUIRibbon.SetRecentItems() 

### ISSUE THREAD

**Issue by ghmrs356** [TurboPack/RibbonFramework#30]
Not sure if I'm doing something wrong, but I crash when I call SetRecentItems() when TUIRibbon attempts to set fRecentItems at line 1311, the actual crash happens in GetCommand() on line 742
```Pascal
function TUIRibbon.GetCommand(pAction: TCustomAction): TUICommand;
var
  lElement: TRibbonMarkupElement;
begin
  if (not Self.Visible) then
    exit(nil);
  for lElement in Self.RibbonMapper do begin  // <<<--- crashes here
    if (lElement.Name = pAction.Name) then begin
      Self.TryGetCommand(lElement.ID, Result);
      Exit(Result);
    end;
  end;
  Exit(nil);
end;
```

because RibbonMapper apparently is nil.

Any ideas?

**Comment by ghmrs356** [comment]
OK, I figured out that I can't call SetRecentItems() before OnLoaded has fired.

However, it still doesn't work (and crashes) because GetCommand returns nil for the recent items action. The action matches the Ribbon command name and it's definitely there (and in the RibbonManager) but TryGetCommand() for the recent element returns nil.

Any ideas why that is? The Recent command is set up in the Ribbon xml correctly.

**Comment by joachimmarder** [comment]
I will extend the TextPad sample to cover that case and see what's wrong.

**Comment by ghmrs356** [comment]
OK thanks! Some more info: I decided to fill the recent list dynamically like you do in the designer's preview. That works but the problem is that ribbon.GetSelectedRecentItem() also blows up because it uses the uninitialized fRecentItems as well. However, I managed to work around that by working my way to the selected recents item in the action event for the recentItemsAction (which is fired, so part of the system works) on foot:
```Pascal
cmd := uiribbonMain.Commands[RibbonMarkup.rcmdApplicationRecent];
if ((cmd <> nil) and (cmd is TUICommandRecentItems)) then begin
    recentItem := TUICommandRecentItemsActionLink(TUICommandRecentItems(cmd).ActionLink).Selected;
    [...]
```
Works and I don't mind maintaining the recent items dynamically. Might be worth investigating the issue though.
